micro-infra-camel/src/main/java/com/ofg/infrastructure/camel/aspects/CorrelationIdOnCamelRouteAspect.java

```diff
@@ -29,12 +29,11 @@ public CorrelationIdOnCamelRouteAspect(UuidGenerator uuidGenerator) {
         correlationIdInterceptor = new CorrelationIdInterceptor(uuidGenerator);
     }
 
-    @Around(value = "execution(* org.apache.camel.builder.RouteBuilder.addRoutesToCamelContext(..))")
-    public Object aroundAddRoutesNoArgs(ProceedingJoinPoint joinPoint) throws Throwable {
+    @Before(value = "execution(* org.apache.camel.builder.RouteBuilder.addRoutesToCamelContext(..))")
+    public void configureCorrelationIdInterceptor(JoinPoint joinPoint) {
         final RouteBuilder targetRouteBuilder = (RouteBuilder) joinPoint.getTarget();
         log.debug("Setting correlationId interception on [{}]", targetRouteBuilder.getClass().getName());
         targetRouteBuilder.interceptFrom().process(correlationIdInterceptor);
         targetRouteBuilder.interceptSendToEndpoint(ANY).process(correlationIdInterceptor);
-        return joinPoint.proceed();
     }
```

Comment:
Btw, @marcingrzejszczak is there a need "clean up" anything in Camel (like in case of threads)?